### PR TITLE
Added Apis For TryGetFullPath

### PR DESCRIPTION
**Pull request by Anipik**
Fixes :- https://github.com/dotnet/corefx/issues/28258
RemoveRelativeSegments Test PR:- https://github.com/dotnet/corefx/pull/28835
Api Test PR :- https://github.com/dotnet/corefx/pull/28837

### CODE REVIEW

**Comment by jkotas** [comment]
`error CS0708: 'TryGetFullPathHelper': cannot declare instance members in a static class`

File: src/mscorlib/shared/System/IO/Path.Unix.cs [hunk 1]
```diff
@@ -44,6 +44,22 @@ public static string GetFullPath(string path)
             return result;
         }
 
+        public static bool TryGetFullPath(ReadOnlySpan<char> path, Span<char> destination, out int charsWritten)
+        {
+            charsWritten = 0;
+
+            if (path.Length == 0 || path.IndexOf('\0') != -1)
```
Comment (jkotas) [inline src/mscorlib/shared/System/IO/Path.Unix.cs]: This return false for both error conditions that won't ever go away; and for buffer-too-small condition.

How do we expect this API to be used? Does the caller have to have hard coded limit on maximum buffer size to try until it declares that it won't work anyway? It does not feel right.

Comment (Anipik) [reply src/mscorlib/shared/System/IO/Path.Unix.cs]: @JeremyKuhne

Comment (danmosemsft) [reply src/mscorlib/shared/System/IO/Path.Unix.cs]: @JeremyKuhne can you help this one along

**Comment by Anipik** [comment]
Closing it until the api gets reviewed again

**Pull request closed without merging**